Source/UnitEx.cpp

```diff
@@ -921,3 +1116,80 @@ void UnitEx::SetAnimation(int animIdx, int animDelay, int animStartDelay, int bo
 
 	func(&(*unitArray)[unitID], 0, animIdx, animDelay, animStartDelay, boolInvisible, boolSkipDoDeath);
 }
+
+int UnitEx::GetNumTruckLoadsSoFar()
+{
+	if (!isInited) {
+		return HFLNOTINITED;
+	}
+
+	BeaconData* p = (BeaconData*)((int)(*unitArray) + (unitID*120) + 0x58);

```

Comment:
Consider using [`uintptr_t`](https://en.cppreference.com/w/c/types/integer) instead of `int` for casting to do address calculations. There are a few occurrences of casts like this below. Actually, this is C++, not C99, so consider using `std::uintptr_t`.

Alternatively, if the `Unit` class was filled out enough, and had `sizeof(Unit) == 120`, you could get the compiler to do much of this calculation for you. With enough defined, you wouldn't even need a cast here.

Reply:
Issue #28 opened.